edgy/core/db/querysets/clauses.py

```diff
@@ -1,24 +1,39 @@
 from typing import Any
 
 import sqlalchemy
+from sqlalchemy.sql.base import ReadOnlyColumnCollection
 
 
-def or_(*args: Any, **kwargs: Any) -> Any:
-    """
-    Creates a SQL Alchemy OR clause for the expressions being passed.
-    """
-    return sqlalchemy.or_(*args, **kwargs)
+class _EnhancedClausesHelper:
+    def __init__(self, op, default_empty):
+        self.op = op
+        self.default_empty = default_empty
 
+    def __call__(self, *args: Any) -> Any:
+        if len(args) == 0:
+            args = [self.default_empty]
+        return self.op(*args)
 
-def and_(*args: Any, **kwargs: Any) -> Any:
-    """
+    def from_kwargs(self, columns_or_model, /, **kwargs: Any) -> Any:
+        if not isinstance(columns_or_model, ReadOnlyColumnCollection) and hasattr(columns_or_model, "columns"):
+            columns_or_model = columns_or_model.columns
+        return self.op(*(getattr(columns_or_model, item[0]) == item[1] for item in kwargs.items()))
+
+or_ = _EnhancedClausesHelper(sqlalchemy.or_, sqlalchemy.false())
+or_.__doc__ = """
+    Creates a SQL Alchemy OR clause for the expressions being passed.
+"""
+and_ = _EnhancedClausesHelper(sqlalchemy.and_, sqlalchemy.true())
+and_.__doc__ = """
     Creates a SQL Alchemy AND clause for the expressions being passed.
-    """
-    return sqlalchemy.and_(*args, **kwargs)
+"""
+
+# alias
```

Comment:
I think this can be removed. I completely forgot to add the Q although everything was there

Reply:
I would keep it. Django users will be otherwise confused what to use to get the same behavior.
Note the subtle difference between `and_()` and `or_()`

Reply:
Ok, I understand your point. Let's keep it but usually this is the reason why we should make the docs better and I speak against myself here